engine/repositories/processor_push.go

```diff
@@ -0,0 +1,99 @@
+package repositories
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+func (s *Service) processPush(op *sdk.Operation) error {
+	gitRepo, path, currentBranch, err := s.processGitClone(op)
+	if err != nil {
+		return sdk.WrapError(err, "unable to process gitclone")
+	}
+	//Check is repo has diverged
+	hasDiverged, err := gitRepo.HasDiverged()
+	if err != nil {
+		log.Error("Repositories> processPush> HasDiverged> [%s] Error: %v", op.UUID, err)
```

Comment:
The error must be handle by the parent func that calls processPush to be logged later instead of being logged then return which mean probably logged two times.

Reply:
err is return to client, that do not need all the stack
`
log.Error("Repositories> processPush> push %s> [%s] error %v", op.Setup.Push.FromBranch, op.UUID, err)
`

Reply:
ok